From 204ed66771fa289319ec40bb477e94a681fec10e Mon Sep 17 00:00:00 2001
From: Vitaly Zaitsev <vitaly@easycoding.org>
Date: Thu, 30 Mar 2017 20:13:48 +0200
Subject: [PATCH 3/4] Build using systemwide cmake.

---
 Telegram/gyp/refresh.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Telegram/gyp/refresh.sh b/Telegram/gyp/refresh.sh
index 9ee05ae..c8689b4 100755
--- a/Telegram/gyp/refresh.sh
+++ b/Telegram/gyp/refresh.sh
@@ -11,9 +11,9 @@ cd $FullScriptPath
 if [ "$MySystem" == "Linux" ]; then
   ../../../Libraries/gyp/gyp --depth=. --generator-output=../.. -Goutput_dir=out Telegram.gyp --format=cmake
   cd ../../out/Debug
-  ../../../Libraries/cmake-3.6.2/bin/cmake .
+  cmake .
   cd ../Release
-  ../../../Libraries/cmake-3.6.2/bin/cmake .
+  cmake .
   cd ../../Telegram/gyp
 else
   #gyp --depth=. --generator-output=../.. -Goutput_dir=out Telegram.gyp --format=ninja
-- 
2.9.3

