From ed276b4ad1dcd1b50af801d83b2ba4befc0099b1 Mon Sep 17 00:00:00 2001
From: Grishka <grishka93@gmail.com>
Date: Tue, 12 Mar 2019 16:34:05 +0300
Subject: [PATCH] I'm too lazy to come up with a commit message

---
 VoIPController.cpp | 2 +-
 VoIPController.h   | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/VoIPController.cpp b/VoIPController.cpp
index 35d3d36..66fb80e 100755
--- a/VoIPController.cpp
+++ b/VoIPController.cpp
@@ -3929,7 +3929,7 @@ void VoIPController::TickJitterBufferAngCongestionControl(){
 
 #pragma mark - Endpoint
 
-Endpoint::Endpoint(int64_t id, uint16_t port, IPv4Address& _address, IPv6Address& _v6address, Type type, unsigned char peerTag[16]) : address(_address), v6address(_v6address){
+Endpoint::Endpoint(int64_t id, uint16_t port, const IPv4Address& _address, const IPv6Address& _v6address, Type type, unsigned char peerTag[16]) : address(_address), v6address(_v6address){
 	this->id=id;
 	this->port=port;
 	this->type=type;
diff --git a/VoIPController.h b/VoIPController.h
index 3add7e9..bb3c30d 100755
--- a/VoIPController.h
+++ b/VoIPController.h
@@ -126,7 +126,7 @@ namespace tgvoip{
 			TCP_RELAY
 		};
 
-		Endpoint(int64_t id, uint16_t port, IPv4Address& address, IPv6Address& v6address, Type type, unsigned char* peerTag);
+		Endpoint(int64_t id, uint16_t port, const IPv4Address& address, const IPv6Address& v6address, Type type, unsigned char* peerTag);
 		Endpoint();
 		~Endpoint();
 		const NetworkAddress& GetAddress() const;
